src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs

```diff
@@ -96,21 +79,45 @@ public void Shutdown(int exitCode = 0)
             }
         }
         
-        
         public int Start(string[] args)
         {
+            _windowOpenedDisposable = Window.WindowOpenedEvent.AddClassHandler(typeof(Window), OnWindowOpened);
```

Comment:
may be single  CompositeDisposable should make it bit easy to handle the dispose

Reply:
Y we can combine them. Will make that change tomorrow or feel free to do it yourself.